overlord/devicestate/devicemgr.go

```diff
@@ -1453,6 +1441,39 @@ func (m *DeviceManager) runFDESetupHook(op string, params *boot.FDESetupHookPara
 	return hookResult, nil
 }
 
+func (m *DeviceManager) checkFDEFeatures(st *state.State) error {
+	// Run fde-setup hook with "op":"features". If the hook
+	// returns any {"features":[...]} reply we consider the
+	// hardware supported. If the hook errors or if it returns
+	// {"error":"hardware-unsupported"} we don't.
+	output, err := m.runFDESetupHook("features", &boot.FDESetupHookParams{})
+	if err != nil {
+		return err
```

Comment:
this error is logged already right?

Reply:
This error will be eventually bubble up into the handler in doSetupRunSystem() and appear as a task error. Do you think we should further log this using a logger?

Reply:
I think it's eaten in checkEncryption actually? that's why I asked about this to help debugging

Reply:
I misread the code, the error was discarded but I pushed a commit that reads it and logs it via a logger.Noticef()

Reply:
Code is updated, I merge this now but happy to do a followup if you want to see tweaks to the message or the way this is done

Reply:
@mvo5 where was the code updated ? I don't see any logger.Noticef in this PR ...